src/leader_scheduler1.rs

```diff
@@ -81,6 +81,12 @@ impl LeaderScheduler1 {
     pub fn next_slot_leader(&self, bank: &Bank) -> Pubkey {
         self.slot_leader_by(bank, Self::next_slot_leader_index)
     }
+
+    // Returns the number of ticks remaining from the specified tick_height to the end of the
+    // slot implied by the tick_height
+    pub fn num_ticks_left_in_slot(bank: &Bank, tick_height: u64) -> u64 {
+        bank.ticks_per_slot() - tick_height % bank.ticks_per_slot() - 1
```

Comment:
Is it possible that all callers pass `bank.tick_height()` into `tick_height`? If so, how about this implementation?

```rust
bank.ticks_per_slot() - bank.tick_index()
```

Reply:
I thought about that. There are cases where tick_height + 1 is being passed in (to calculate slot for the next tick).

Reply:
Could those be replaced with `next_slot_leader()`?